classes/Kohana/Jam/Field/Integer.php

```diff
@@ -32,5 +32,36 @@ public function set(Jam_Validated $model, $value, $is_changed)
 
 		return $value;
 	}
+	
+	/**
+	 * Like Jam_Field::_default, but it won't convert 0 and 0.0 to empty_value
+	 *
+	 * @param  Jam_Validated $model
+	 * @param  mixed $value
+	 * @return array 1st element the converted value; 2nd element a boolean
+	 * indicating if the value should not be processed further
+	 */
+	protected function _default(Jam_Validated $model, $value)
+	{
+		$return = FALSE;
+
+		$value = $this->run_filters($model, $value);
+
+		// Convert empty values to NULL, if needed
+		if ($this->convert_empty AND empty($value) AND $value !== 0 AND $value !== 0.0 AND $value !== "0" AND $value !== "0.0")
```

Comment:
Yes, but following that path if you provide an empty string for the value it will still be converted to 0 rather than NULL... which is not what I want. However this brings up the other question which is the correct behaviour in that case... '' -> 0 or '' -> NULL for an integer field?